src/main.c

```diff
@@ -480,26 +344,31 @@ static int start_component(zl_comp_t *comp) {
 
   int fd_count = 3;
   int fd_map[3];
+  char bin[PATH_MAX];
 
-  if (strcmp(&comp->bin[bin_len - 3], ".sh") == 0) {
-    script = fopen(full_path, "r");
-    if (script == NULL) {
-      ERROR("script not open for %s - %s\n", comp->name, strerror(errno));
-      return -1;
-    }
-    fd_map[0] = dup(fileno(script));
-    fclose(script);
-  } else {
-    fd_map[0] = dup(STDIN_FILENO);
+  snprintf(bin, sizeof(bin), "%s/bin/internal/start-component-with-launcher.sh", getenv("ROOT_DIR"));
```

Comment:
I think we can use `start-component.sh` directly here. The extra source lines in `start-component-with-launcher.sh` are already included in `prepare-environment.sh` and will be sourced in `start-component.sh`.

Reply:
I see, yes it looks like we can simplify due to https://github.com/zowe/zowe-install-packaging/blob/master/bin/internal/start-component.sh#L54

Reply:
ok. I changed the source code to use `start-component.sh` directly.